src/override/malloc.cc

```diff
@@ -40,10 +40,10 @@ extern "C"
   {
     bool overflow = false;
     size_t sz = bits::umul(size, nmemb, overflow);
-    if (overflow)
+    if (overflow || ((nmemb | size) == (size_t)-1))
```

Comment:
I guess you are required to fail sensibly, if you are given negative numbers?  Or just -1.  Does this cover all the cases that are specified?

Reply:
glibc sets ENOMEM if either nmemb or size is -1. I assume that it would be unreasonable for an application to rely on this behavior for other values, but maybe that could also be said for -1. Is there a more reasonable upper bound that we could put on the size of allocations?

Reply:
It seems unreasonable to even rely on it for -1 to me.  I am just concerned about putting tests on a critical path.